src/modules/commander/mag_calibration.cpp

```diff
@@ -626,6 +626,138 @@ calibrate_return mag_calibrate_all(orb_advert_t *mavlink_log_pub, int32_t cal_ma
+					if (worker_data.calibration[cur_mag].device_id() != 0) {
+						for (unsigned i = 0; i < worker_data.calibration_counter_total[cur_mag]; i++) {
+
+							const Vector3f sample{worker_data.x[cur_mag][i], worker_data.y[cur_mag][i], worker_data.z[cur_mag][i]};
+
+							float scale_data[9] {
+								diag[cur_mag](0),    offdiag[cur_mag](0), offdiag[cur_mag](1),
+								offdiag[cur_mag](0),    diag[cur_mag](1), offdiag[cur_mag](2),
+								offdiag[cur_mag](1), offdiag[cur_mag](2),    diag[cur_mag](2)
+							};
+
+							// apply calibration
+							const Vector3f m{Matrix3f{scale_data} *(sample - sphere[cur_mag])};
+
+							// store back in worker_data
+							worker_data.x[cur_mag][i] = m(0);
+							worker_data.y[cur_mag][i] = m(1);
+							worker_data.z[cur_mag][i] = m(2);
+						}
+					}
+				}
+
+				// rotate internal mag data to board
+				const Dcmf board_rotation = calibration::GetBoardRotation();
+
+				for (unsigned i = 0; i < worker_data.calibration_counter_total[internal_index]; i++) {
+
+					const Vector3f m = board_rotation * Vector3f{worker_data.x[internal_index][i],
+							worker_data.y[internal_index][i], worker_data.z[internal_index][i]};
+
+					worker_data.x[internal_index][i] = m(0);
+					worker_data.y[internal_index][i] = m(1);
+					worker_data.z[internal_index][i] = m(2);
+				}
+
+				for (unsigned cur_mag = 0; cur_mag < MAX_MAGS; cur_mag++) {
+					if (worker_data.calibration[cur_mag].external() && (worker_data.calibration[cur_mag].device_id() != 0)) {
+
+						const int last_sample_index = math::min(worker_data.calibration_counter_total[internal_index],
+											worker_data.calibration_counter_total[cur_mag]);
+
+						float diff_sum[ROTATION_MAX] {};
+
+						float min_diff = FLT_MAX;
+						Rotation best_rotation = ROTATION_NONE;
+
+						for (int r = ROTATION_NONE; r < ROTATION_MAX; r++) {
+							for (int i = 0; i < last_sample_index; i++) {
+
+								float x = worker_data.x[cur_mag][i];
+								float y = worker_data.y[cur_mag][i];
+								float z = worker_data.z[cur_mag][i];
+								rotate_3f((enum Rotation)r, x, y, z);
+
+								Vector3f diff = Vector3f{x, y, z} - Vector3f{worker_data.x[internal_index][i], worker_data.y[internal_index][i], worker_data.z[internal_index][i]};
+
+								diff_sum[r] += diff.norm();
+							}
+
+							if (diff_sum[r] < min_diff) {
+								min_diff = diff_sum[r];
+								best_rotation = (Rotation)r;
+							}
+						}
+
+
+						// Check that the best rotation is at least twice as good as the next best
+						bool smallest_check_passed = true;
+
+						for (int r = ROTATION_NONE; r < ROTATION_MAX; r++) {
+							if (r != best_rotation) {
+								if (diff_sum[r] < (min_diff * 2.f)) {
+									smallest_check_passed = false;
+								}
+							}
+						}
+
+
+						// Check that the average error across all samples (relative to internal mag) is less than the minimum earth field (~ 0.25 Gauss)
+						const float mag_error_ga = (min_diff / last_sample_index);
```

Comment:
```suggestion
						const float mag_error_gs = (min_diff / last_sample_index);
```
https://en.wikipedia.org/wiki/Gauss_(unit)

Reply:
Thanks, TIL.